### PR TITLE
displaying alert for invalid or no config for push and keycloak services

### PR DESCRIPTION
**Pull request by pawelpaszki**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-3061

## Description

An alert is now being displayed, when displaying authenticate pages with invalid or missing Keycloak configuration and when accessing Push page without Push service being set up.

## Progress

- [x] Display alerts for keycloak and push config

## Verification:

remove Keycloak config from mobile-services.json, launch the app and from the menu select Authentication, Identity Management or Push Notifications to display the alert.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -45,6 +45,9 @@
     "cordova-plugin-dialogs": "^2.0.1",
     "cordova-plugin-inappbrowser": "^3.0.0",
     "cordova-plugin-ionic-webview": "^1.2.1",
+    "cordova-plugin-iroot": "^0.7.0",
```
Comment (wtrocki) [inline package.json]: Remove this!

**Comment by wtrocki** [comment]
@pawelpaszki Great job. This resolves this bug as well: https://issues.jboss.org/browse/AEROGEAR-3282

File: package.json [hunk 2]
```diff
@@ -45,6 +45,8 @@
     "cordova-plugin-dialogs": "^2.0.1",
     "cordova-plugin-inappbrowser": "^3.0.0",
     "cordova-plugin-ionic-webview": "^1.2.1",
+    "cordova-plugin-is-debug": "^1.0.0",
+    "cordova-plugin-pincheck": "0.0.5",
```
Comment (ciaranRoche) [inline package.json]: These two plugins can be removed.

**Comment by pawelpaszki** [comment]
@wtrocki @ciaranRoche I did not add those packages to package.json. Perhaps they are added during the build. I will keep an eye on those when completing the work for this pull request

**Comment by ciaranRoche** [comment]
@pawelpaszki I know, it happens we merged a change that should of fixed that problem, so I would think go ahead and clean the project, remove the likes of your `node_modules`, `plugins` folder etc, and run a clean build, it should stop the likes of those plugins being added to the `package.json`

File: src/pages/auth/auth.ts [hunk 3]
```diff
@@ -29,6 +31,27 @@ export class AuthPage {
...
+            if (configKeys.length <= 1 || configKeys.indexOf('url') === -1) {
+                this.configValid = false;
+                this.showInvalidAuthConfigAlert();
+            } else {
+                this.configValid = true;
+            }
         }
     }
+
+    showInvalidAuthConfigAlert(): void {
```
Comment (ciaranRoche) [inline src/pages/auth/auth.ts]: I was wondering, would it be better if we created a validate service for these functions, and then just call on them where we need them.

What do you think @pawelpaszki @StephenCoady

Comment (pawelpaszki) [reply src/pages/auth/auth.ts]: @ciaranRoche you are probably right. I will make the required changes

File: src/mobile-services.json [hunk 4]
```diff
@@ -4,23 +4,7 @@
   "namespace": "myproject",
   "clientId": "example_client_id",
   "services": [
-    {
-      "id": "keycloak",
```
Comment (StephenCoady) [inline src/mobile-services.json]: re add this

File: src/pages/auth/auth.ts [hunk 5]
```diff
@@ -3,8 +3,10 @@ import { NavController, NavParams } from 'ionic-angular';
 import { authProvider } from '../../services/auth.service';
 import { Auth } from '@aerogear/auth';
 import { ToastController } from 'ionic-angular';
-
+import { Dialogs } from '@ionic-native/dialogs';
```
Comment (StephenCoady) [inline src/pages/auth/auth.ts]: remove unused import

File: src/pages/deviceTrust/deviceTrust.ts [hunk 6]
```diff
@@ -2,8 +2,8 @@ import { Component } from '@angular/core';
 import { NavController } from 'ionic-angular';
 import { SecurityService, SecurityCheckType, SecurityCheckResult } from '@aerogear/security';
 import { Dialogs } from '@ionic-native/dialogs';
```
Comment (StephenCoady) [inline src/pages/deviceTrust/deviceTrust.ts]: unused import

File: src/pages/push/push.ts [hunk 7]
```diff
@@ -1,3 +1,4 @@
+import { AlertService } from '../../services/alert.service';
```
Comment (StephenCoady) [inline src/pages/push/push.ts]: unused as we are missing the logic on this page to check

**Comment by pawelpaszki** [comment]
@ciaranRoche @StephenCoady I can't squash there using rebase, because there are too many confilicts, but I think I have found another way. WIll ping you, once successful

**Pull request merged**